packages/vm/src/instance.rs

```diff
@@ -58,38 +58,38 @@ where
                 //   value region too small: -1_000_001
                 //   key does not exist: -1_001_001
                 // Ownership of both input and output pointer is not transferred to the host.
-                "db_read" => Func::new(move |ctx: &mut Ctx, key_ptr: u32, value_ptr: u32| -> i32 {
+                "db_read" => Func::new(move |ctx: &mut Ctx, key_ptr: u32, value_ptr: u32| -> VmResult<i32> {
```

Comment:
Nice to see this signature working. But what happens on an error?

Reply:
Wasmer halts the contract execution and returns the error we provided. I have not handled the errors yet on the side that calls into the contract.
See here: https://github.com/wasmerio/wasmer/issues/1426

Reply:
Cool, thanks